messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/WildFlyBindingRegistry.java

```diff
@@ -91,16 +93,24 @@ public void unbind(String name) {
             return;
         }
         // remove the binding service
-        bindingService.setMode(ServiceController.Mode.REMOVE);
-        final StabilityMonitor monitor = new StabilityMonitor();
-        monitor.addController(bindingService);
+        final CountDownLatch latch = new CountDownLatch(1);
+        bindingService.addListener(new LifecycleListener() {
+            @Override
+            public void handleEvent(final ServiceController<?> controller, final LifecycleEvent event) {
+                if (event == LifecycleEvent.REMOVED) {
+                    ROOT_LOGGER.unboundJndiName(bindInfo.getAbsoluteJndiName());
+                    latch.countDown();
+                }
+            }
+        });
         try {
-            monitor.awaitStability();
-            ROOT_LOGGER.unboundJndiName(bindInfo.getAbsoluteJndiName());
-        } catch (InterruptedException e) {
-            ROOT_LOGGER.failedToUnbindJndiName(name, 5, SECONDS.toString().toLowerCase(Locale.US));
+            bindingService.setMode(ServiceController.Mode.REMOVE);
         } finally {
-            monitor.removeController(bindingService);
+            try {
+                latch.await(5, SECONDS);
```

Comment:
Before this would wait until interrupted; now it has a fairly short 5 sec timeout. 

History shows that originally it was a 5 sec wait but a change you made in Jan 2014 made it indefinite. At this point I'm concerned that we have almost 7 years of bake with it this way, vs changing it back and hoping that not blocking is ok and that 5 secs is still a valid number.

Reply:
I returned it back to previous undefinite value.